db/db_options_test.cc

```diff
@@ -117,6 +117,77 @@ TEST_F(DBOptionsTest, GetLatestCFOptions) {
             GetMutableCFOptionsMap(dbfull()->GetOptions(handles_[1])));
 }
 
+TEST_F(DBOptionsTest, SetBytesPerSync) {
+  Options options;
+  options.create_if_missing = true;
+  options.bytes_per_sync = 512 * 512;
+  options.use_direct_reads = false;
+  options.env = env_;
+  Reopen(options);
+  int counter = 0;
+  int low_bytes_per_sync = 0;
+  int i = 0;
+  const std::string kValue(1024 * 1024 * 8, 'v');
+  ASSERT_EQ(512 * 512, dbfull()->GetDBOptions().bytes_per_sync);
+  rocksdb::SyncPoint::GetInstance()->SetCallBack(
+      "WritableFileWriter::RangeSync:0", [&](void* arg) {
+        counter++;
+      });
+  rocksdb::SyncPoint::GetInstance()->EnableProcessing();
+  WriteOptions write_opts;
+  for (; i < 10; i++) {
+    Put(Key(i), kValue, write_opts);
+  }
+  Flush();
+  i = 0;
+  low_bytes_per_sync = counter;
+  counter = 0;
+  ASSERT_OK(dbfull()->SetDBOptions({{"bytes_per_sync", "1505728"}}));
+  ASSERT_EQ(1505728, dbfull()->GetDBOptions().bytes_per_sync);
+  for (; i < 10; i++) {
+    Put(Key(i), kValue, write_opts);
+  }
+  Flush();
+  ASSERT_GT(counter, 0);
+  ASSERT_GT(low_bytes_per_sync, 0);
+  ASSERT_GT(low_bytes_per_sync, counter);
```

Comment:
Trigger a manual compaction (i.e. `ASSERT_OK(dbfull()->CompactRange(CompactRangeOptions(), nullptr, nullptr)))`) and check counter.